src/univariate/continuous/rician.jl

```diff
@@ -0,0 +1,175 @@
+    Rician(T(d.ν), T(d.σ); check_args=false)
+end
+
+#### Parameters
+
+shape(d::Rician) = d.ν^2 / (2 * d.σ^2)
+scale(d::Rician) = d.ν^2 + 2 * d.σ^2
+
+params(d::Rician) = (d.ν, d.σ)
+partype(d::Rician{T}) where {T<:Real} = T
+
+#### Statistics
+
+# helper
+_Lhalf(x) = exp(x/2) * ((1-x) * besseli(zero(x), -x/2) - x * besseli(oneunit(x), -x/2))
+
+mean(d::Rician{T}) where T = d.σ * √T(π/2) * _Lhalf(-d.ν^2/(2 * d.σ^2))
+var(d::Rician) = 2 * d.σ^2 + d.ν^2 - (π * d.σ^2 / 2) * _Lhalf(-d.ν^2/(2 * d.σ^2))^2
+
+function mode(d::Rician)
+    m = mean(d)
+    _minimize_gss(x -> -pdf(d, x), zero(m), m)
+end
+
+# helper: 1D minimization using Golden-section search
+function _minimize_gss(f, a, b; tol=1e-12)
+    ϕ = (√5 + 1) / 2
+    c = b - (b - a) / ϕ
+    d = a + (b - a) / ϕ
+    while abs(b - a) > tol
+        if f(c) < f(d)
+            b = d
+        else
+            a = c
+        end
+        c = b - (b - a) / ϕ
+        d = a + (b - a) / ϕ
+    end
+    (b + a) / 2
+end
+
+#### PDF/CDF/quantile delegated to NoncentralChisq
+
+# NOTE:
+# 1.0ν and 1.0x used to support x::Float32, since the underlying NoncentralChisq delegates
+# to StatsFuns(rmath.jl) which doesn't have support for Float32
+
+function quantile(d::Rician, x::Real)
+    ν, σ = params(d)
+    return sqrt(quantile(NoncentralChisq(2, (1.0ν / σ)^2), 1.0x)) * σ
+end
+
+function cquantile(d::Rician, x::Real)
+    ν, σ = params(d)
+    return sqrt(cquantile(NoncentralChisq(2, (1.0ν / σ)^2), 1.0x)) * σ
+end
+
+function pdf(d::Rician, x::Real)
+    (x ≤ 0 || isinf(x)) && return zero(1.0x)
+    ν, σ = params(d)
+    return 2 * x / σ^2 * pdf(NoncentralChisq(2, (1.0ν / σ)^2), (1.0x / σ)^2)
+end
+
+function logpdf(d::Rician, x::Real)
+    (x ≤ 0 || isinf(x)) && return oftype(1.0x, -Inf)
+    ν, σ = params(d)
+    return log(2 * x / σ^2) + logpdf(NoncentralChisq(2, (1.0ν / σ)^2), (1.0x / σ)^2)
+end
+
+function cdf(d::Rician, x::Real)
+    x ≤ 0 && return zero(1.0x)
+    ν, σ = params(d)
+    return cdf(NoncentralChisq(2, (1.0ν / σ)^2), (1.0x / σ)^2)
+end
+
+function logcdf(d::Rician, x::Real)
+    x ≤ 0 && return oftype(1.0x, -Inf)
+    ν, σ = params(d)
+    return logcdf(NoncentralChisq(2, (1.0ν / σ)^2), (1.0x / σ)^2)
+end
+
+#### Sampling
+
+function rand(rng::AbstractRNG, d::Rician)
+    x = randn(rng) * d.σ + d.ν
+    y = randn(rng) * d.σ
+    hypot(x, y)
+end
+
+#### Fitting
+
+# implementation based on the Koay inversion technique
+function fit(::Type{<:Rician}, x::AbstractArray{T}; tol=1e-12, maxiters=500) where T
+    μ₁ = mean(x)
+    μ₂ = var(x)
+    r = μ₁ / √μ₂
+    if r < sqrt(π/(4-π))
+        ν = zero(float(T))
+        σ = scale(fit(Rayleigh, float.(x)))
+    else
+        ξ(θ) = 2 + θ^2 - π/8 * exp(-θ^2 / 2) * ((2 + θ^2) * besseli(0, θ^2 / 4) + θ^2 * besseli(1, θ^2 / 4))^2
+        g(θ) = sqrt(ξ(θ) * (1+r^2) - 2)
+        θ = 1.0
+        for j ∈ 1:maxiters
+            θ⁻ = θ
+            θ = g(θ)
+            abs(θ - θ⁻) < tol && break
+        end
```

Comment:
The type of `θ` might change in the first step and hence create a type instability. I'd suggest
```suggestion
        θ = g(1)
        for j in 1:maxiters
            θ⁻ = θ
            θ = g(θ)
            abs(θ - θ⁻) < tol && break
        end
```